include/maths/CSignal.h

```diff
@@ -418,6 +418,16 @@ class MATHS_EXPORT CSignal {
     static std::size_t selectComponentSize(const TFloatMeanAccumulatorVec& values,
                                            std::size_t period);
 
+    //! Apply a smooth resampling of \p component at \p size granularity.
+    //!
+    //! \param[in] size The granularity at which to resample.
```

Comment:
If you don't think it's worth making the method more defensive then you could specifically note here that zero doesn't make sense and will cause a crash.